src/Views/AppListView.vala

```diff
@@ -103,26 +121,21 @@ namespace AppCenter.Views {
                 return p1.is_plugin ? 1 : -1;
             }
 
+            int sp1 = search_priority (p1.get_name ());
+            int sp2 = search_priority (p2.get_name ());
+            if (sp1 != sp2) {
+                return sp2 - sp1;
+            }
+
             return p1.get_name ().collate (p2.get_name ());
         }
 
-#if CURATED
         [CCode (instance_pos = -1)]
         protected override int package_row_compare (Widgets.AppListRow row1, Widgets.AppListRow row2) {
-            bool p1_is_elementary_native = row1.get_package ().is_native;
-            bool p1_is_plugin = row1.get_package ().is_plugin;
-
-            if (p1_is_elementary_native != row2.get_package ().is_native) {
-                return p1_is_elementary_native ? -1 : 1;
-            }
-
-            if (p1_is_plugin != row2.get_package ().is_plugin) {
-                return p1_is_plugin ? 1 : -1;
-            }
-
-            return base.package_row_compare (row1, row2);
+            return compare_packages (row1.get_package (), row2.get_package ());
```

Comment:
This code removal should probably have happened regardless of having the search sorting added. Without it, the list_store and list_box could be sorted differently - leading to resorting when more results are shown.